### PR TITLE
Refactor `_adapt_lights` into `_update_manual_control_and_maybe_adapt`

### PR DESCRIPTION
**Pull request by th3w1zard1**
Adaptive Lighting originally had two different methods:
`_adapt_lights`
and
`_adapt_light`
both called functions relating to manual control. Seemed messy to me so I moved all of the manual control stuff to `_adapt_lights` then renamed it to `_update_manual_control_and_maybe_adapt` to suit the format.

### CODE REVIEW

File: tests/test_switch.py [hunk 1]
```diff
@@ -198,7 +199,8 @@ async def setup_lights_and_switch(hass, extra_conf=None):
             CONF_SUNSET_TIME: datetime.time(SUNSET.hour),
             CONF_INITIAL_TRANSITION: 0,
             CONF_TRANSITION: 0,
-            CONF_DETECT_NON_HA_CHANGES: True,
+            CONF_DETECT_NON_HA_CHANGES: False,
+            CONF_TAKE_OVER_CONTROL: True,
```
Comment (basnijholt) [inline tests/test_switch.py:203]: Why change these defaults for all tests?

**Comment by basnijholt** [comment]
Did you change the permissions on service.yaml on purpose? (100644 → 100755)

File: tests/test_switch.py (hunk 1 above)
Comment (th3w1zard1) [reply tests/test_switch.py:203]: I changed them to be certain my test worked as I'm unfamiliar with a lot of the other code.

**Comment by th3w1zard1** [comment]
> Did you change the permissions on service.yaml on purpose? (100644 → 100755)

I'm not sure what you're asking

**Comment by basnijholt** [comment]
OK doesn't matter! If you look at the diff on GitHub you see:
image

File: custom_components/adaptive_lighting/switch.py [hunk 2]
```diff
@@ -1648,6 +1658,10 @@ def __init__(self, hass: HomeAssistant):
         # Track light transitions
         self.transition_timers: dict[str, _AsyncSingleShotTimer] = {}
 
+        # When a state is different `max_cnt_significant_changes` times in a row,
+        # mark it as manually_controlled.
+        self.max_cnt_significant_changes = 2
```
Comment (basnijholt) [inline custom_components/adaptive_lighting/switch.py:1663]: Are you adding this back? Or accidental?

Comment (th3w1zard1) [reply custom_components/adaptive_lighting/switch.py:1663]: Accidental, happened during the merge!

**Review by basnijholt — DISMISSED** [review]

**Review by basnijholt — APPROVED** [review]

**Pull request merged**